web/concrete/blocks/form/controller.php

```diff
@@ -581,11 +581,17 @@ public function action_submit_form($bID = false)
                 if ($this->redirectCID > 0) {
                     $pg = Page::getByID($this->redirectCID);
                     if (is_object($pg) && $pg->cID) {
-                        $this->redirect($pg->getCollectionPath());
+                        $targetPage = $pg;
                     }
                 }
-                $c = Page::getCurrentPage();
-                header("Location: ".Core::make('helper/navigation')->getLinkToCollection($c, true)."?surveySuccess=1&qsid=".$this->questionSetId."#formblock".$this->bID);
+                if (is_object($targetPage)) {
```

Comment:
Here `$targetPage` may not be defined, thus raising a PHP warning.
What about initializing ` $targetPage` to `null`?